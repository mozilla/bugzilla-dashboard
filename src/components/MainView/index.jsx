import React from 'react';
import PropTypes from 'prop-types';
import BugzillaComponents from '../BugzillaComponents';
import Reportees from '../Reportees';

class MainView extends React.Component {
  renderTabContents() {
    const {
      ldapEmail, partialOrg, onPersonDetails, teamComponents,
      bugzillaComponents, onComponentDetails, selectedTabIndex,
    } = this.props;
    switch (selectedTabIndex) {
      case 0: {
        return (
          <Reportees
            ldapEmail={ldapEmail}
            partialOrg={partialOrg}
            onPersonDetails={onPersonDetails}
          />
        );
      }
      case 1: {
        return (
          <BugzillaComponents
            bugzillaComponents={teamComponents}
            onComponentDetails={onComponentDetails}
          />
        );
      }
      case 2: {
        return (
          <BugzillaComponents
            bugzillaComponents={bugzillaComponents}
            onComponentDetails={onComponentDetails}
          />
        );
      }
      default: {
        return null;
      }
    }
  }

  render() {
    return this.renderTabContents();
  }
}

MainView.propTypes = {
  ldapEmail: PropTypes.string.isRequired,
  partialOrg: PropTypes.shape({}).isRequired,
  bugzillaComponents: PropTypes.arrayOf(PropTypes.shape({})),
  teamComponents: PropTypes.arrayOf(PropTypes.shape({})),
  onComponentDetails: PropTypes.func.isRequired,
  onPersonDetails: PropTypes.func.isRequired,
  selectedTabIndex: PropTypes.number.isRequired,
};

MainView.defaultProps = {
  bugzillaComponents: [],
  teamComponents: [],
};

export default MainView;
