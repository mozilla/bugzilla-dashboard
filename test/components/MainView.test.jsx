import React from 'react';
import renderer from 'react-test-renderer';
import MainView from '../../src/components/MainView';
import partialOrg from '../mocks/partialOrg';
import bugzillaComponents from '../mocks/bugzillaComponents';

it('renders Someone with no reportees', () => {
  const tree = renderer
    .create((
      <MainView
        ldapEmail="someone@mozilla.com"
        partialOrg={partialOrg}
        bugzillaComponents={bugzillaComponents}
      />
    ))
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders Manager who has reportees', () => {
  const tree = renderer
    .create((
      <MainView
        ldapEmail="manager@mozilla.com"
        partialOrg={partialOrg}
        bugzillaComponents={bugzillaComponents}
      />
    ))
    .toJSON();
  expect(tree).toMatchSnapshot();
});
