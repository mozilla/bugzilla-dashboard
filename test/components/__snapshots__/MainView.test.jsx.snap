// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Manager who has reportees 1`] = `
<div>
  <h2
    className="MainView-header-2"
  >
    Manager
  </h2>
  <div
    className="MainView-content-1"
  >
    <div
      className="Reportees-root-3"
    >
      <h3
        className="Reportees-header-4"
      >
        Reportees
      </h3>
      <div
        height="1rem"
      >
         
      </div>
      <div
        className="Reportees-person-6"
      >
        <span
          name="manager@mozilla.com"
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex="0"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-7 Reportees-icon-5"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
          </svg>
        </span>
        <span>
          Manager 
        </span>
      </div>
      <div
        className="Reportees-person-6"
      >
        <span
          name="someone@mozilla.com"
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex="0"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-7 Reportees-icon-5"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
          </svg>
        </span>
        <span>
          Someone 
        </span>
      </div>
    </div>
    <div>
      <div>
        <h3
          className="BugzillaComponents-header-16"
        >
          Components
        </h3>
        <table>
          <thead>
            <tr>
              <th />
              <th />
              <th
                className="BugzillaComponents-metricLabel-18"
              >
                Untriaged
              </th>
              <th
                className="BugzillaComponents-metricLabel-18"
              >
                Needinfo
              </th>
              <th
                className="BugzillaComponents-metricLabel-18"
              >
                P1s
              </th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>
                <div
                  name="Core::DOM: IndexedDB"
                  onClick={[Function]}
                  onKeyPress={[Function]}
                  role="button"
                  tabIndex="0"
                >
                  <svg
                    aria-hidden="true"
                    className="MuiSvgIcon-root-7 BugzillaComponents-icon-19"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                    />
                    <path
                      d="M0 0h24v24H0z"
                      fill="none"
                    />
                  </svg>
                </div>
              </td>
              <td>
                Core::DOM: IndexedDB
              </td>
            </tr>
            <tr>
              <td>
                <div
                  name="Core::JavaScript Engine"
                  onClick={[Function]}
                  onKeyPress={[Function]}
                  role="button"
                  tabIndex="0"
                >
                  <svg
                    aria-hidden="true"
                    className="MuiSvgIcon-root-7 BugzillaComponents-icon-19"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                    />
                    <path
                      d="M0 0h24v24H0z"
                      fill="none"
                    />
                  </svg>
                </div>
              </td>
              <td>
                Core::JavaScript Engine
              </td>
            </tr>
            <tr>
              <td>
                <div
                  name="Core::DOM: Core & HTML"
                  onClick={[Function]}
                  onKeyPress={[Function]}
                  role="button"
                  tabIndex="0"
                >
                  <svg
                    aria-hidden="true"
                    className="MuiSvgIcon-root-7 BugzillaComponents-icon-19"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                    />
                    <path
                      d="M0 0h24v24H0z"
                      fill="none"
                    />
                  </svg>
                </div>
              </td>
              <td>
                Core::DOM: Core & HTML
              </td>
              <td
                className="BugzillaComponents-metric-17"
              >
                <a
                  href="https://bugzilla.mozilla.org/buglist.cgi?component=DOM%3A%20Core%20%26%20HTML&f1=bug_severity&f2=keywords&f3=resolution&limit=0&o1=notequals&o2=notsubstring&o3=isempty&product=Core&v1=enhancement&v2=meta"
                >
                  944
                </a>
              </td>
            </tr>
            <tr>
              <td>
                <div
                  name="Toolkit::Async Tooling"
                  onClick={[Function]}
                  onKeyPress={[Function]}
                  role="button"
                  tabIndex="0"
                >
                  <svg
                    aria-hidden="true"
                    className="MuiSvgIcon-root-7 BugzillaComponents-icon-19"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                    />
                    <path
                      d="M0 0h24v24H0z"
                      fill="none"
                    />
                  </svg>
                </div>
              </td>
              <td>
                Toolkit::Async Tooling
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders Manager who has reportees and teams 1`] = `
<div>
  <h2
    className="MainView-header-2"
  >
    Manager
  </h2>
  <div
    className="MainView-content-1"
  >
    <div
      className="Reportees-root-3"
    >
      <h3
        className="Reportees-header-4"
      >
        Reportees
      </h3>
      <div
        height="1rem"
      >
         
      </div>
      <div
        className="Reportees-person-6"
      >
        <span
          name="manager@mozilla.com"
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex="0"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-7 Reportees-icon-5"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
          </svg>
        </span>
        <span>
          Manager 
        </span>
      </div>
      <div
        className="Reportees-person-6"
      >
        <span
          name="someone@mozilla.com"
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex="0"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-7 Reportees-icon-5"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
          </svg>
        </span>
        <span>
          Someone 
        </span>
      </div>
    </div>
    <div>
      <div>
        <h3
          className="BugzillaComponents-header-16"
        >
          Components
        </h3>
        <table>
          <thead>
            <tr>
              <th />
              <th />
              <th
                className="BugzillaComponents-metricLabel-18"
              >
                Untriaged
              </th>
              <th
                className="BugzillaComponents-metricLabel-18"
              >
                Needinfo
              </th>
              <th
                className="BugzillaComponents-metricLabel-18"
              >
                P1s
              </th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>
                <div
                  name="Core::DOM: IndexedDB"
                  onClick={[Function]}
                  onKeyPress={[Function]}
                  role="button"
                  tabIndex="0"
                >
                  <svg
                    aria-hidden="true"
                    className="MuiSvgIcon-root-7 BugzillaComponents-icon-19"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                    />
                    <path
                      d="M0 0h24v24H0z"
                      fill="none"
                    />
                  </svg>
                </div>
              </td>
              <td>
                Core::DOM: IndexedDB
              </td>
            </tr>
            <tr>
              <td>
                <div
                  name="Core::JavaScript Engine"
                  onClick={[Function]}
                  onKeyPress={[Function]}
                  role="button"
                  tabIndex="0"
                >
                  <svg
                    aria-hidden="true"
                    className="MuiSvgIcon-root-7 BugzillaComponents-icon-19"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                    />
                    <path
                      d="M0 0h24v24H0z"
                      fill="none"
                    />
                  </svg>
                </div>
              </td>
              <td>
                Core::JavaScript Engine
              </td>
            </tr>
            <tr>
              <td>
                <div
                  name="Core::DOM: Core & HTML"
                  onClick={[Function]}
                  onKeyPress={[Function]}
                  role="button"
                  tabIndex="0"
                >
                  <svg
                    aria-hidden="true"
                    className="MuiSvgIcon-root-7 BugzillaComponents-icon-19"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                    />
                    <path
                      d="M0 0h24v24H0z"
                      fill="none"
                    />
                  </svg>
                </div>
              </td>
              <td>
                Core::DOM: Core & HTML
              </td>
              <td
                className="BugzillaComponents-metric-17"
              >
                <a
                  href="https://bugzilla.mozilla.org/buglist.cgi?component=DOM%3A%20Core%20%26%20HTML&f1=bug_severity&f2=keywords&f3=resolution&limit=0&o1=notequals&o2=notsubstring&o3=isempty&product=Core&v1=enhancement&v2=meta"
                >
                  944
                </a>
              </td>
            </tr>
            <tr>
              <td>
                <div
                  name="Toolkit::Async Tooling"
                  onClick={[Function]}
                  onKeyPress={[Function]}
                  role="button"
                  tabIndex="0"
                >
                  <svg
                    aria-hidden="true"
                    className="MuiSvgIcon-root-7 BugzillaComponents-icon-19"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                    />
                    <path
                      d="M0 0h24v24H0z"
                      fill="none"
                    />
                  </svg>
                </div>
              </td>
              <td>
                Toolkit::Async Tooling
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders Someone with no reportees 1`] = `
<div>
  <h2
    className="MainView-header-2"
  >
    Someone
  </h2>
  <div
    className="MainView-content-1"
  >
    <div
      className="Reportees-root-3"
    >
      <h3
        className="Reportees-header-4"
      >
        Reportees
      </h3>
      <div
        height="1rem"
      >
         
      </div>
      <div
        className="Reportees-person-6"
      >
        <span
          name="manager@mozilla.com"
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex="0"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-7 Reportees-icon-5"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
          </svg>
        </span>
        <span>
          Manager 
        </span>
      </div>
      <div
        className="Reportees-person-6"
      >
        <span
          name="someone@mozilla.com"
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex="0"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-7 Reportees-icon-5"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
          </svg>
        </span>
        <span>
          Someone 
        </span>
      </div>
    </div>
    <div>
      <div>
        <h3
          className="BugzillaComponents-header-16"
        >
          Components
        </h3>
        <table>
          <thead>
            <tr>
              <th />
              <th />
              <th
                className="BugzillaComponents-metricLabel-18"
              >
                Untriaged
              </th>
              <th
                className="BugzillaComponents-metricLabel-18"
              >
                Needinfo
              </th>
              <th
                className="BugzillaComponents-metricLabel-18"
              >
                P1s
              </th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>
                <div
                  name="Core::DOM: IndexedDB"
                  onClick={[Function]}
                  onKeyPress={[Function]}
                  role="button"
                  tabIndex="0"
                >
                  <svg
                    aria-hidden="true"
                    className="MuiSvgIcon-root-7 BugzillaComponents-icon-19"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                    />
                    <path
                      d="M0 0h24v24H0z"
                      fill="none"
                    />
                  </svg>
                </div>
              </td>
              <td>
                Core::DOM: IndexedDB
              </td>
            </tr>
            <tr>
              <td>
                <div
                  name="Core::JavaScript Engine"
                  onClick={[Function]}
                  onKeyPress={[Function]}
                  role="button"
                  tabIndex="0"
                >
                  <svg
                    aria-hidden="true"
                    className="MuiSvgIcon-root-7 BugzillaComponents-icon-19"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                    />
                    <path
                      d="M0 0h24v24H0z"
                      fill="none"
                    />
                  </svg>
                </div>
              </td>
              <td>
                Core::JavaScript Engine
              </td>
            </tr>
            <tr>
              <td>
                <div
                  name="Core::DOM: Core & HTML"
                  onClick={[Function]}
                  onKeyPress={[Function]}
                  role="button"
                  tabIndex="0"
                >
                  <svg
                    aria-hidden="true"
                    className="MuiSvgIcon-root-7 BugzillaComponents-icon-19"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                    />
                    <path
                      d="M0 0h24v24H0z"
                      fill="none"
                    />
                  </svg>
                </div>
              </td>
              <td>
                Core::DOM: Core & HTML
              </td>
              <td
                className="BugzillaComponents-metric-17"
              >
                <a
                  href="https://bugzilla.mozilla.org/buglist.cgi?component=DOM%3A%20Core%20%26%20HTML&f1=bug_severity&f2=keywords&f3=resolution&limit=0&o1=notequals&o2=notsubstring&o3=isempty&product=Core&v1=enhancement&v2=meta"
                >
                  944
                </a>
              </td>
            </tr>
            <tr>
              <td>
                <div
                  name="Toolkit::Async Tooling"
                  onClick={[Function]}
                  onKeyPress={[Function]}
                  role="button"
                  tabIndex="0"
                >
                  <svg
                    aria-hidden="true"
                    className="MuiSvgIcon-root-7 BugzillaComponents-icon-19"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                    />
                    <path
                      d="M0 0h24v24H0z"
                      fill="none"
                    />
                  </svg>
                </div>
              </td>
              <td>
                Toolkit::Async Tooling
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</div>
`;
